packages/story-editor/src/app/media/utils/useMediaInfo.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo } from '@googleforcreators/react';
+import {
+  getTimeTracker,
+  trackError,
+  trackEvent,
+} from '@googleforcreators/tracking';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../config';
+import {
+  MEDIA_VIDEO_DIMENSIONS_THRESHOLD,
+  MEDIA_VIDEO_FILE_SIZE_THRESHOLD,
+} from '../../../constants';
+
+function loadScriptOnce(url) {
+  if (document.querySelector(`script[src="${url}"]`)) {
+    return Promise.resolve();
+  }
+
+  return new Promise((resolve, reject) => {
+    const script = document.createElement('script');
+    script.async = true;
+    script.crossOrigin = 'anonymous';
+    script.src = url;
+    script.addEventListener('load', resolve);
+    script.addEventListener('error', reject);
+    document.head.appendChild(script);
+  });
+}
+
+/**
+ * @typedef MediaInfo
+ * @property {string} mimeType File mime type.
+ * @property {number} fileSize File size in bytes.
+ * @property {string} format File format.
+ * @property {string} codec File codec.
+ * @property {number} frameRate Frame rate (rounded).
+ * @property {number} height Height in px.
+ * @property {number} width Width in px.
+ * @property {string} colorSpace Color space.
+ * @property {number} duration Video duration.
+ * @property {string} videoCodec Video codec.
+ * @property {string} audioCodec Audio codec.
+ * @property {boolean} isMuted Whether the video is muted.
+ */
+
+/**
+ * Custom hook to interact with mediainfo.js.
+ *
+ * @see https://mediainfo.js.org/
+ * @return {Object} Functions and vars related to mediainfo.js usage.
+ */
+function useMediaInfo() {
+  const { mediainfoUrl } = useConfig();
+
+  const getFileInfo = useCallback(
+    /**
+     * Returns information about a given media file.
+     *
+     * @param {File} file File object.
+     * @return {Promise<MediaInfo|null>} File info or null on error.
+     */
+    async (file) => {
+      const getSize = () => file.size;
+
+      const readChunk = (chunkSize, offset) =>
```

Comment:
Can we reuse `createFileReader` function?

Reply:
Should this function, look like this. 

```
const createFileReader = (file) => {
  const reader = new window.FileReader();
  return new Promise((resolve, reject) => {
    reader.onload =  (event) => {
            if (event.target.error) {
              reject(event.target.error); 
           }
         resolve(reader);
    }
    reader.onerror = reject;
    reader.readAsArrayBuffer(file);
  });
};
```

Reply:
At the very least we should also have this line 

```
reader.onerror = reject;
``` 

Maybe add a todo, saying to use this function at a later date.